datafusion/sql/src/select.rs

```diff
@@ -574,10 +574,19 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
         empty_from: bool,
         planner_context: &mut PlannerContext,
     ) -> Result<Vec<Expr>> {
-        projection
-            .into_iter()
-            .map(|expr| self.sql_select_to_rex(expr, plan, empty_from, planner_context))
-            .collect::<Result<Vec<Expr>>>()
+        let mut prepared_select_exprs = vec![];
+        let mut errors = vec![];
+        for expr in projection {
+            match self.sql_select_to_rex(expr, plan, empty_from, planner_context) {
+                Ok(expr) => prepared_select_exprs.push(expr),
+                Err(err) => errors.push(err),
+            }
+        }
+        if !errors.is_empty() {
```

Comment:
It would be nice to only wrap with a DataFusionError::Collection if this actually more than one error

Maybe you could make some sort of builder like
```rust
let mut builder = DataFusionErrorBuilder::new();
... 
let builder = builder.add_error(err)
...
builder.build()?; // returns Err(DataFusionError if any have been pushed)
Ok(prepared_select_exprs)
```

And handle the unwapping there

Reply:
Ah yes good point! I like that, will change